app/controllers/api/v1/search_controller.rb

```diff
@@ -0,0 +1,289 @@
+              property :type do
+                key :type, :string
+                key :description, 'Type of search performed'
+              end
+              property :terms do
+                key :type, :string
+                key :title, 'Keywords used in search'
+              end
+              property :page do
+                key :type, :string
+                key :title, 'Pagination control'
+              end
+              property :facets do
+                key :type, :array
+                key :title, 'SearchFacets'
+                key :description, 'Array of facets/filters used in search'
+                items do
+                  key :type, :object
+                  key :title, 'SearchFacet'
+                  property :id do
+                    key :type, :string
+                    key :description, 'ID of facet'
+                  end
+                  property :filters do
+                    key :type, :array
+                    key :description, 'Matching filters'
+                    items do
+                      key :type, :object
+                      key :title, 'Filter'
+                      property :name do
+                        key :type, :string
+                        key :description, 'Display value of filter'
+                      end
+                      property :id do
+                        key :type, :string
+                        key :description, 'ID value of filter'
+                      end
+                    end
+                  end
+                end
+              end
+              property :studies do
+                key :type, :array
+                items do
+                  key :title, 'Study, StudyFiles'
+                  key :'$ref', :SiteStudyWithFiles
+                end
+              end
+            end
+          end
+          response 406 do
+            key :description, 'Accept or Content-Type headers missing or misconfigured'
+          end
+          response 500 do
+            key :description, 'Server error'
+          end
+        end
+      end
+
+      def index
+        order = [:view_order.asc, :name.asc]
+        @viewable = Study.viewable(current_api_user).order_by(order)
+
+        # if search params are present, filter accordingly
+        if !params[:terms].blank?
+          search_terms = sanitize_search_values(params[:terms])
+          # determine if search values contain possible study accessions
+          possible_accessions = StudyAccession.sanitize_accessions(search_terms.split)
+          @studies = @viewable.any_of({:$text => {:$search => search_terms}}, {:accession.in => possible_accessions})
+        else
+          @studies = @viewable
+        end
+
+        # only call BigQuery if list of possible studies is larger than 0 and we have matching facets to use
+        if @studies.count > 0 && @facets.any?
+          @big_query_search = Api::V1::SearchController.generate_bq_query_string(@facets)
```

Comment:
I'd use 'self.class.generate_bq_query_string(@facets)'  to call the method, it's shorter and robust to file renames